### DISCUSSION TITLE
Typing "pipe" ( | ) in sql editor put the focus elsewhere

### DISCUSSION THREAD

**Discussion by Thomas-LEILLET-Sanofi** [dbeaver/dbeaver#20310] [Q&A] [answered]
### Description

When typing | (pipe) , it puts the focus on date-time field editor if open below, or just I-dont-know-where on the window - but impossible to type Pipe.

Observed in 23.1.0.202306041918, not observed in version 23.0.4.202305071722 to which I downgraded back.

### DBeaver Version

Version 23.1.0.202306041918

**Comment by ShadelessFox** [comment]
Hello @Thomas-LEILLET-Sanofi,

1. How do you type that character exactly? Are you using the combination <kbd>SHIFT+\\</kbd>?
2. What's your OS?
3. What's your keyboard layout?

Additionally, footage showing the issue would be helpful.

**Comment by cvillerm** [comment]
I have the exact same issue with v23.1.0.202306041918 installed on Windows 10, using an AZERTY keyboard layout (the pipe character is accessible using "AtlGr-6"). I don't have any similar issue with other AltGr-accessible characters

**Comment by Thomas-LEILLET-Sanofi** [comment]
> I have the exact same issue with v23.1.0.202306041918 installed on Windows 10, using an AZERTY keyboard layout (the pipe character is accessible using "AtlGr-6"). I don't have any similar issue with other AltGr-accessible characters

![image](https://github.com/dbeaver/dbeaver/assets/130557869/97ab9b85-721a-4a56-92af-d25bcd773ee0)


(win11 , French AZERTY keyboard too here)

**Comment by ShadelessFox** [comment, marked as answer]
Thanks for the explanation.

I cannot reproduce the issue myself by pressing the same keys. But, according to your description, I suspect you're accidentally executing the "Toggle results panel" action that reveals the results panel if it was hidden. I'm not sure why that is the case, but I assume Eclipse treats `AltGr` as `Ctrl` in some situations, but why it doesn't occur in older versions is a mystery to me.

Please try changing the following keybinding to see if it helps:

![image](https://github.com/dbeaver/dbeaver/assets/35821147/ccf368cd-bd7d-4952-8bd1-c8cb8e8fb500)

**Comment by cvillerm** [comment]
Thanks for the hint. Here is the actual faulting binding:
![image](https://github.com/dbeaver/dbeaver/assets/9128625/7ad4d957-7f68-4419-a30b-d300ce0f6bca)
Unbinding it makes it work again.

**Comment by ShadelessFox** [comment]
Hooray!

Waiting for @Thomas-LEILLET-Sanofi to confirm that the provided solutions works.

**Comment by Thomas-LEILLET-Sanofi** [comment]
![image](https://github.com/dbeaver/dbeaver/assets/130557869/aed8a053-fb43-41ba-b9cb-e64fa11ee98a)

**Comment by MrThibal** [comment]
Thanks for the solution !
I hope we won't have to use it soon.
